src/acquisition/covidcast/database.py

```diff
@@ -92,110 +92,141 @@ def count_all_rows(self):
         `value` = VALUES(`value`),
         `stderr` = VALUES(`stderr`),
         `sample_size` = VALUES(`sample_size`)
     '''
+    zero_is_latest_issue_sql = f'''
+      UPDATE
+      (
+        SELECT DISTINCT `source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`
+        FROM `{tmp_table_name}`
+      ) AS TMP
+      LEFT JOIN `covidcast`
+      USING (`source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`)
+      SET `is_latest_issue`=0
+    '''
+    set_is_latest_issue_sql = f'''
+      UPDATE 
+      (
+        SELECT `source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`, MAX(`issue`) AS `issue`
+        FROM
+        (
+          SELECT DISTINCT `source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value` 
+          FROM `{tmp_table_name}`
+        ) AS TMP
+        LEFT JOIN `covidcast`
+        USING (`source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`)
+        GROUP BY `source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`
+      ) AS TMP
+      LEFT JOIN `covidcast`
+      USING (`source`, `signal`, `time_type`, `geo_type`, `time_value`, `geo_value`, `issue`)
+      SET `is_latest_issue`=1        
+    '''
 
-    args = (
-      source,
-      signal,
-      time_type,
-      geo_type,
-      time_value,
-      geo_value,
-      value,
-      stderr,
-      sample_size,
-      issue,
-      lag
-    )
+    # TODO: ^ do we want to reset `direction_updated_timestamp` and `direction` in the duplicate key case?
 
-    self._cursor.execute(sql, args)
+    # TODO: consider handling cc_rows as a generator instead of a list
+    self._cursor.execute(create_tmp_table_sql)
+
+    try:
+
+      num_rows = len(cc_rows)
+      total = 0
+      if not batch_size:
+        batch_size = num_rows
+      num_batches = ceil(num_rows/batch_size)
+      for batch_num in range(num_batches):
+        start = batch_num * batch_size
+        end = min(num_rows, start + batch_size)
+        length = end - start
+
+        args = [(
+          row.source,
+          row.signal,
+          row.time_type,
+          row.geo_type,
+          row.time_value,
+          row.geo_value,
+          row.value,
+          row.stderr,
+          row.sample_size,
+          row.issue,
+          row.lag
+        ) for row in cc_rows[start:end]]
+
+
+        result = self._cursor.executemany(insert_into_tmp_sql, args)
+        self._cursor.execute(insert_or_update_sql)
+        self._cursor.execute(zero_is_latest_issue_sql)
+        self._cursor.execute(set_is_latest_issue_sql)
+        self._cursor.execute(truncate_tmp_table_sql)
+
+        if result is None:
+          # the SQL connector does not support returning number of rows affected
+          total = None
+        else:
+          total += result
+        if commit_partial:
+          self._connection.commit()
+    except Exception as e:
+      print('AAA')
```

Comment:
you can get rid of this, but its really not necessary

Reply:
It was removed in the previous commit